src/Data/Fin/Permutation.agda

```diff
@@ -246,3 +287,30 @@ lift₀-transpose i j (suc k) with does (k ≟ i)
 ... | false with does (k ≟ j)
 ...   | false = refl
 ...   | true = refl
+
+insert-punchIn : ∀ i j (π : Permutation m n) k → insert i j π ⟨$⟩ʳ punchIn i k ≡ punchIn j (π ⟨$⟩ʳ k)
+insert-punchIn i j π k with i ≟ punchIn i k
+... | yes i≡punchInᵢk = ⊥-elim (punchInᵢ≢i i k (sym i≡punchInᵢk))
+... | no  i≢punchInᵢk = begin
+  punchIn j (π ⟨$⟩ʳ punchOut i≢punchInᵢk) ≡⟨ cong (λ l → punchIn j (π ⟨$⟩ʳ l)) (punchOut-cong i refl) ⟩
+  punchIn j (π ⟨$⟩ʳ punchOut (punchInᵢ≢i i k ∘ sym)) ≡⟨ cong (λ l → punchIn j (π ⟨$⟩ʳ l)) (punchOut-punchIn i) ⟩
+  punchIn j (π ⟨$⟩ʳ k) ∎
+
+insert-remove : ∀ i (π : Permutation (suc m) (suc n)) → insert i (π ⟨$⟩ʳ i) (remove i π) ≈ π
+insert-remove {m = m} {n = n} i π j with i ≟ j
+... | yes i≡j = cong (π ⟨$⟩ʳ_) i≡j
+... | no  i≢j = begin
+  punchIn (π ⟨$⟩ʳ i) (punchOut (punchInᵢ≢i i (punchOut i≢j) ∘ sym ∘ Inverse.injective π)) ≡⟨ punchIn-punchOut _ ⟩
+  π ⟨$⟩ʳ punchIn i (punchOut i≢j) ≡⟨ cong (π ⟨$⟩ʳ_) (punchIn-punchOut i≢j) ⟩
+  π ⟨$⟩ʳ j ∎
+
+remove-insert : ∀ i j (π : Permutation m n) → remove i (insert i j π) ≈ π
+remove-insert i j π k with i ≟ i
+... | yes _ = begin
+  punchOut {i = j} _
+    ≡⟨ punchOut-cong j (insert-punchIn i j π k) ⟩
+  punchOut {i = j} (punchInᵢ≢i j (π ⟨$⟩ʳ k) ∘ sym)
+    ≡⟨ punchOut-punchIn j ⟩
+  π ⟨$⟩ʳ k
+    ∎
+... | no i≢i = ⊥-elim (i≢i refl)
```

Comment:
Can we use `contradiction` here instead? Same for other uses of `⊥-elim`. Also the proof would be slightly more readable if you put the `no` above the `yes` in my opinion.